blackfire/blackfire.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Blackfire extension stubs.
+ * @link https://blackfire.io
+ */
+class BlackfireProbe
+{
+    /**
+     * Returns a global singleton and enables it by default.
+     *
+     * @return \BlackfireProbe
+     */
+    public static function getMainInstance() {}
+
+    /**
+     * Tells whether the probe is currently profiling or not.
+     *
+     * @return bool
+     */
+    public static function isEnabled() {}
+
+    /**
+     * Adds a marker for the Timeline View.
+     *
+     * @param string $markerName
+     */
+    public static function addMarker($markerName = '') {}
+
+    /**
+     * Tells if the probe is cryptographically verified,
+     * i.e. if the signature in X-Blackfire-Query HTTP header or
+     * BLACKFIRE_QUERY environment variable is valid.
+     *
+     * @return bool
+     */
+    public function isVerified() {}
+
+    /**
+     * @param string $configuration
+     */
+    public function setConfiguration($configuration) {}
```

Comment:
This param is nullable if this is the source: https://github.com/blackfireio/php-sdk/blob/4188f61bad20b45b19b61c3b6225f9aec2c8c8e4/src/BlackfireProbe.php#L291

Reply:
It is not nullable in the probe itself:

```
Extension [ <persistent> extension #68 blackfire version 1.31.0~mac-x64-non_zts73 ] {
  - Classes [2] {
    Class [ <internal:blackfire> final class BlackfireProbe ] {

      - Methods [10] {
        Method [ <internal:blackfire, ctor> public method __construct ] {

          - Parameters [4] {
            Parameter #0 [ <required> $query ]
            Parameter #1 [ <optional> $probeId ]
            Parameter #2 [ <optional> $probeToken ]
            Parameter #3 [ <optional> $outputUrl ]
          }
        }

        Method [ <internal:blackfire> public method isVerified ] {
        }

        Method [ <internal:blackfire> public method getResponseLine ] {
        }

        Method [ <internal:blackfire> public method setConfiguration ] {

          - Parameters [1] {
            Parameter #0 [ <required> $str ]
          }
        }

        Method [ <internal:blackfire> public method enable ] {
        }

        Method [ <internal:blackfire> public method disable ] {
        }

        Method [ <internal:blackfire> public method close ] {
        }

        Method [ <internal:blackfire> public method discard ] {
        }

        Method [ <internal:blackfire> public method getHeapUsage ] {
        }

        Method [ <internal:blackfire> public method createSubProfileQuery ] {
        }
      }
    }
  }
}
```

Reply:
I mean that by that polyfill' source code - 
method setConfiguration accepts null argument: `$probe->setConfiguration(null);`,
it can be indicated in the doc: `@param string|null`
But I don't see the real source code, so it was an assumption.

Reply:
Sure, no problem 😉 ! It is true that the source code is not public, that's why I pasted a snippet of `php --re blackfire` result.